### DISCUSSION TITLE
UGS multiple machine profiles possible

### DISCUSSION THREAD

**Discussion by zutim13** [grblHAL/core#407] [Q&A]
Hey there, hey @terjeio!
My name is Tim and i talked to grblgru about a good solution for my machines. He told me to ask you here on github, maybe you can help me. As i use MacOS i would like to use UGS but i have 3 different machines and wonder if i can load different machine profiles in UGS to use only one Computer and not 3.
I will work with fusion 360 as the 3d builder and also for the cam. i would like to use DM856 stepper drivers, because i would like to have nearly noisless stepper motors. So i wonder whats the best way to get there. i found out 32 Bit might be the solution, but i wonder which microcontroller and hardware i should use, also which breakout board. i have 3 axis.
Would be cool if you could help me! 
Thanks a lot,
Tim

**Comment by terjeio** [comment]
> As i use MacOS i would like to use UGS but i have 3 different machines and wonder if i can load different machine profiles

I do not use UGS myself so I do not know, check the [UGS site](https://winder.github.io/ugs_website/) or ask at the [github repo](https://github.com/winder/Universal-G-Code-Sender).

> i wonder which microcontroller and hardware i should use, also which breakout board.

There are several [controllers](https://github.com/grblHAL/Controllers) that can be used with external stepper drivers: PicoCNC, Pico BOB, Flexi-HAL, T41U5XBB, GRBLHAL2000, E5XMCS_T41 and others.

**Reply by zutim13** [reply]
Hey Torje, thanks for your reply!
Is there another G-CODE Sender you can recommend which have the possibility to have different machine files?
Thanks for your time!
Tim

**Reply by terjeio** [reply]
> Is there another G-CODE Sender you can recommend which have the possibility to have different machine files?

I do not keep track of sender capabilities so I am sorry I do not know.

For Windows [ioSender](https://github.com/terjeio/ioSender) can be started with different profiles and it also configures itself from data read from the controller.

**Reply by karoria** [reply]
You can try gSender which supports grblHAL and is availabpe for mac too. I don't know if it can run simultaneous 3 machine profiles. You can try it. Other thing you can try is use webUI in case you are OK with basic interface without graphics display. That way you can run machines from your browser only so it will become OS independent.

**Reply by zutim13** [reply]
ah, ok, thank you, i will try that too!
